src/core/net/ip6.cpp

```diff
@@ -654,8 +654,8 @@ Error Ip6::HandleFragment(Message &aMessage)
     offset          = FragmentHeader::FragmentOffsetToBytes(fragmentHeader.GetOffset());
     payloadFragment = aMessage.GetLength() - aMessage.GetOffset() - sizeof(fragmentHeader);
 
-    LogInfo("Fragment with id %u received > %u bytes, offset %u", (unsigned)fragmentHeader.GetIdentification(),
-            (unsigned)payloadFragment, (unsigned)offset);
+    LogInfo("Fragment with id %lu received > %lu bytes, offset %lu", ToUlong(fragmentHeader.GetIdentification()),
+            ToUlong(payloadFragment), ToUlong(offset));
```

Comment:
Suggest the following.

```suggestion
    LogInfo("Fragment with id %lu received > %u bytes, offset %u", ToUlong(fragmentHeader.GetIdentification()),
            payloadFragment, offset);
```

Since `payloadFragment` and `offset` are `uint16_t` and `%u` works fine for `uint16_t`.
We only need to fix the `fragmentHeader.GetIdentification()` which is `uint32_t`.

Reply:
that makes sense. updated.